plugin/pkg/global/resourcereferencemanager/admission.go

```diff
@@ -534,22 +560,43 @@ func (r *ReferenceManager) ensureProjectNamespace(project *core.Project) error {
 	return nil
 }
 
-func (r *ReferenceManager) ensureSecretBindingReferences(ctx context.Context, attributes admission.Attributes, binding *core.SecretBinding) error {
+func (r *ReferenceManager) ensureBindingReferences(ctx context.Context, attributes admission.Attributes, binding runtime.Object) error {
+	var (
+		quotas          []corev1.ObjectReference
+		secretNamespace string
+		secretName      string
+	)
+	switch attributes.GetKind().GroupKind() {
+	case core.Kind("SecretBinding"):
+		b := binding.(*core.SecretBinding)
```

Comment:
It is very unlikely, but the conversion to SecretBinding may fail, hence it should be checked before proceeding. 

```suggestion
		b, ok := binding.(*core.SecretBinding)
		if !ok {
			return errors.New("failed to convert binding to SecretBinding")
		}
```